### PR TITLE
Pact API improvements 

### PR DESCRIPTION
**Pull request by slpopejoy**
Ready once Pact PR goes in

### CODE REVIEW

File: src/Chainweb/Pact/Types.hs [hunk 1]
```diff
@@ -140,7 +140,7 @@ data PactDbStatePersist = PactDbStatePersist
     , _pdbspPactDbState :: PactDbState
     }
 
-newtype GasSupply = GasSupply { _gasSupply :: Decimal }
+newtype GasSupply = GasSupply { _gasSupply :: GasPrice }
```
Comment (emilypi) [inline src/Chainweb/Pact/Types.hs]: `GasSupply` is our current working total. Why was `Decimal` replaced with `GasPrice`?

Comment (slpopejoy) [reply src/Chainweb/Pact/Types.hs]: Maybe not the best idea, it was in hopes of minimizing coercion. I'll look at this

Comment (slpopejoy) [reply src/Chainweb/Pact/Types.hs]: Ah the other reason is it needs a JSON representation to go into the message body, ie in `mkBuyGasCmd` and `GasPrice` has the right instances.

The larger issue is using dedicated newtypes for gas prices, when in reality these are simply money values which don't have a first class representation in the API. I propose leaving this for now and addressing in a Pact PR, ie a `Price` or `Money` API type.

Comment (slpopejoy) [reply src/Chainweb/Pact/Types.hs]: OK figured out a better solution, and one we should have done for `GasPrice`: newtype `ParsedDecimal` to grab its instances. Likewise, `GasLimit` should newtype `ParsedInteger`.

File: stack.yaml [hunk 2]
```diff
@@ -40,6 +40,6 @@ extra-deps:
   - git: https://github.com/kadena-io/merkle-log.git
     commit: a7ae61d7082afe3aa1a0fd0546fc1351a2f7c376
   - git: https://github.com/kadena-io/pact.git
-    commit: 9456a750f67b62d48bed5ef2e60fbc7e9d2a4396
+    commit: b5601c601c3e497c938f6d804eb21f9a3fb30a2a
```
Comment (fosskers) [inline stack.yaml]: This is the commit from the PR. The merged commit is: 0915aff9eed762b2ee58c648d900bc619e6fff96

File: cabal.project [hunk 3]
```diff
@@ -9,7 +9,7 @@ package aeson
 source-repository-package
     type: git
     location: https://github.com/kadena-io/pact.git
-    tag: 9456a750f67b62d48bed5ef2e60fbc7e9d2a4396
+    tag: b5601c601c3e497c938f6d804eb21f9a3fb30a2a
```
Comment (fosskers) [inline cabal.project]: This is the commit from the PR. The merged commit is: 0915aff9eed762b2ee58c648d900bc619e6fff96

File: default.nix [hunk 4]
```diff
@@ -200,8 +200,8 @@ in
         pact = dontCheck ( addBuildDepend (self.callCabal2nix "pact" (pkgs.fetchFromGitHub {
           owner = "kadena-io";
           repo = "pact";
-          rev = "9456a750f67b62d48bed5ef2e60fbc7e9d2a4396";
-          sha256 = "039fmgb5d4p524vagmkmmp3gajmvrly7r3iwy9mcdg6j7a6vqj9w";
+          rev = "b5601c601c3e497c938f6d804eb21f9a3fb30a2a";
```
Comment (fosskers) [inline default.nix]: This is the commit from the PR. The merged commit is: 0915aff9eed762b2ee58c648d900bc619e6fff96

File: src/Chainweb/Pact/Types.hs (hunk 1 above)
Comment (emilypi) [reply src/Chainweb/Pact/Types.hs]: Works for me

**Pull request merged**